test/dotcom/system_status/groups_test.exs

```diff
@@ -5,177 +5,182 @@ defmodule Dotcom.SystemStatus.GroupsTest do
-               },
-               %{
-                 route_id: "Green",
-                 sub_routes: [],
-                 statuses: [%{description: "Normal Service", time: nil}]
                }
              ] =
                Groups.groups(
                  [
                    Alert.build(:alert,
                      effect: :delay,
-                     informed_entity: [InformedEntity.build(:informed_entity, route: "Orange")],
-                     active_period: [current_active_period()]
+                     informed_entity: [
+                       InformedEntity.build(:informed_entity, route: affected_route_id)
+                     ],
+                     active_period: [{alert_start_time, alert_end_time}]
                    )
                  ],
-                 now()
+                 time
                )
+               |> statuses_for(affected_route_id)
+    end
 
-      assert [
-               %{
-                 route_id: "Blue",
-                 sub_routes: [],
-                 statuses: [%{description: "Shuttle Buses", time: nil}]
-               },
-               %{
-                 route_id: "Orange",
-                 sub_routes: [],
-                 statuses: [%{description: "Normal Service", time: nil}]
-               },
-               %{
-                 route_id: "Red",
-                 sub_routes: [],
-                 statuses: [%{description: "Normal Service", time: nil}]
-               },
-               %{
-                 route_id: "Green",
-                 sub_routes: [],
-                 statuses: [%{description: "Normal Service", time: nil}]
-               }
-             ] =
-               Groups.groups(
-                 [
-                   Alert.build(:alert,
-                     effect: :shuttle,
-                     informed_entity: [InformedEntity.build(:informed_entity, route: "Blue")],
-                     active_period: [current_active_period()]
-                   )
-                 ],
-                 now()
-               )
+    test "when there's an alert for a heavy rail line, shows 'Normal Service' for the other lines" do
+      affected_route_id = Faker.Util.pick(@heavy_rail_lines)
+
+      time = time_today()
+      alert_start_time = time_before(time)
+      alert_end_time = time_after(time)
+
+      groups =
+        Groups.groups(
+          [
+            Alert.build(:alert,
+              effect: :delay,
+              informed_entity: [InformedEntity.build(:informed_entity, route: affected_route_id)],
+              active_period: [{alert_start_time, alert_end_time}]
+            )
+          ],
+          time
+        )
+
+      @heavy_rail_lines
+      |> List.delete(affected_route_id)
+      |> Enum.each(fn route_id ->
+        assert [
+                 %{
+                   statuses: [%{time: nil, description: "Normal Service"}],
+                   sub_routes: [],
+                   route_id: ^route_id
+                 }
```

Comment:
suggestion(non-blocking): This could read real nice broken out into a function, like `Enum.all?(lines, &shows_normal_service?/1)` which validates the nil time and "Normal Service" text (though I suppose you'd want a separate check to verify the correct route IDs involved, maybe)

Reply:
Question: I've given the tests a pretty thorough once-over (actually more like a twice- or thrice- over). Do you think this is still necessary?